src/plugins/range/range.c

```diff
@@ -0,0 +1,272 @@
+/**
+ * @file
+ *
+ * @brief Source for range plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ *
+ */
+
+#include "range.h"
+#include <ctype.h>
+#include <errno.h>
+#include <kdbhelper.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+int elektraRangeOpen (Plugin * handle ELEKTRA_UNUSED, Key * errorKey ELEKTRA_UNUSED)
+{
+	// plugin initialization logic
+	// this function is optional
+
+	return 1; // success
+}
+
+int elektraRangeClose (Plugin * handle ELEKTRA_UNUSED, Key * errorKey ELEKTRA_UNUSED)
+{
+	// free all plugin resources and shut it down
+	// this function is optional
+
+	return 1; // success
+}
+
+static int rangeStringToRange (const char * rangeString, long long int * min, long long int * max)
+{
+	int factorA = 1;
+	int factorB = 1;
+	long long int a, b;
+	a = b = 0;
+	int pos = 0;
+
+	const char * ptr = rangeString;
+	while (*ptr)
+	{
+		if (isspace (*ptr))
+		{
+			++ptr;
+			continue;
+		}
+		else if (*ptr == '-')
+		{
+			if (pos == 0)
+			{
+				if (factorA == -1)
+				{
+					return -1;
+				}
+				factorA = -1;
+			}
+			else if (pos == 1)
+			{
+				pos = 2;
+			}
+			else if (pos == 2)
+			{
+				if (factorB == -1)
+				{
+					return -1;
+				}
+				factorB = -1;
+			}
+			else
+			{
+				return -1;
+			}
+			++ptr;
+			continue;
+		}
+		else if (isdigit (*ptr))
+		{
+			if (pos == 0)
+			{
+				pos = 1;
+				char * endPtr;
+				a = strtoll (ptr, &endPtr, 10);
```

Comment:
i don't think there's much gain in it, but probably the way the current version handles it (union + enum) ?